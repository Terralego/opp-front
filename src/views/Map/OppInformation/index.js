import OppInformation from './OppInformation';

export default OppInformation;
