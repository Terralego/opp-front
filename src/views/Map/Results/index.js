import { withTranslation } from 'react-i18next';

import { Results } from './Results';
import { connectAppProvider } from '../../../AppLayout/context';

export default connectAppProvider({
  isResultUnfold: 'isResultUnfold',
  filteredViewpoints: 'filteredViewpoints.current',
  toggleResultFoldedState: 'actions.toggleResultFoldedState',
  getPaginatedFilteredViewpoints: 'actions.getPaginatedFilteredViewpoints',
  onMouseEnterResult: 'actions.onMouseEnterResult',
  onMouseLeaveResult: 'actions.onMouseLeaveResult',
})(withTranslation()(Results));
