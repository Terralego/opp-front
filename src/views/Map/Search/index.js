import { withTranslation } from 'react-i18next';

import { Search } from './Search';
import { connectAppProvider } from '../../../AppLayout/context';

export default connectAppProvider({
  map: 'map',
  isSearchUnfold: 'isSearchUnfold',
  toggleSearchFoldedState: 'actions.toggleSearchFoldedState',
})(withTranslation()(Search));
