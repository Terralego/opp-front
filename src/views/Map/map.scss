.map {
  display: flex;
  position: relative;
}
