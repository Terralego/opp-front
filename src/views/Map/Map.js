import React, { useEffect } from 'react';

import Visualizer from '../../components/Visualizer';
import OppInformation from './OppInformation';
import Search from './Search';
import Results from './Results';

import './map.scss';

export const Map = ({ isNavBarVisible, toggleNavBar }) => {
  useEffect(() => {
    !isNavBarVisible && toggleNavBar();
  }, [isNavBarVisible, toggleNavBar]);

  return (
    <div className="map">
      <Search />
      <Results />
      <Visualizer />
      <OppInformation />
    </div>
  );
};

export default Map;
