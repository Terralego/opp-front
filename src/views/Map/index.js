import { connectAppProvider } from '../../AppLayout/context';
import Map from './Map';

export default connectAppProvider({
  isNavBarVisible: 'isNavBarVisible',
  toggleNavBar: 'actions.toggleNavBar',
})(Map);
