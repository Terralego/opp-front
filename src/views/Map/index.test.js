import React from 'react';
import renderer from 'react-test-renderer';
import Map from './Map';

jest.mock('../../components/Visualizer', () => () => <p>VisualizerMock</p>);
jest.mock('./OppInformation', () => () => <p>OppInformationMock</p>);
jest.mock('./Search', () => () => <p>Search</p>);
jest.mock('./Results', () => () => <p>Results</p>);

it('should render', () => {
  const tree = renderer.create(<Map />).toJSON();
  expect(tree).toMatchSnapshot();
});
