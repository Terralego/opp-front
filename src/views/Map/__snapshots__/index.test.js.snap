// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="map"
>
  <p>
    Search
  </p>
  <p>
    Results
  </p>
  <p>
    VisualizerMock
  </p>
  <p>
    OppInformationMock
  </p>
</div>
`;
